daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/processors/parsers/SeparatedSequenceChildParseResultHelper.scala

```diff
@@ -62,6 +62,7 @@ object SeparatedSequenceChildBehavior {
   sealed abstract class PositionalLike extends Type
   sealed abstract class PositionalTrailing extends PositionalLike
   case object Positional extends PositionalLike
```

Comment:
Positional means that the way you tell what element/child of a sequence you are dealing with is by counting which child it is.  Only true when things are required, or have fixed number of children, excepting the last declared in the sequence, which can be optional, because we know it can't be anything else. 

Non-Positional means the data has some way of identifying what child e.g., by initiators. In other words, it's ambiguous until you look at the data which it is. Hence, one ends up speculating, and trying to parse a child, and when it succeeds, then it "is one" and when it fails it isn't and you go on to the next declared child if it is optional.

Reply:
Makes sense. I see know that the comment above these enums does say something to that effect.

What you describe sounds a lot like this new `PositionalNever` thing. I wonder if `PositionalNever` shouldn't be required and `Positional` should have the `arrayCompleteChecks`, but that there are some other bugs with `Positional`? I can't devise any tests that seem to work in correctly, so maybe we commit this as is, and eventually a bug will crop up that reveals if there is an actual issue?